src/runtime/builder.rs

```diff
@@ -0,0 +1,107 @@
+use runtime::inner::Inner;
+use runtime::runtime_impl::Runtime;
+
+use reactor::Reactor;
+
+use std::io;
+
+use tokio_threadpool::Builder as ThreadPoolBuilder;
+
+
+
+/// Builds Tokio Runtime with custom configuration values.
+///
+/// Methods can be chanined in order to set the configuration values. The
+/// Runtime is constructed by calling [`build`].
+///
+/// New instances of `Builder` are obtained via [`Builder::new`].
+///
+/// See function level documentation for details on the various configuration
+/// settings.
+///
+/// [`build`]: #method.build
+/// [`Builder::new`]: #method.new
+///
+/// # Examples
+///
+/// ```
+/// # extern crate tokio;
+/// # extern crate tokio_threadpool;
+/// # use tokio::runtime::Builder;
+///
+/// # pub fn main() {
+/// let runtime = Builder::new()
+///     .threadpool_builder(
+///         // Create a thread pool with some configuration values
+///         tokio_threadpool::Builder::new()
+///             .name_prefix("my-runtime-worker-")
+///             .pool_size(4)
+///             .clone()
+///         )
+///     .build();
+/// // ... call runtime.run(...)
+/// # let _ = runtime;
+/// # }
+/// ```
+#[derive(Debug, Clone)]
```

Comment:
It would be better to leave `Clone` off as this adds additional constraints in the future.

Reply:
Done